frontend/public/metalkube/components/host/host-detail.tsx

```diff
@@ -12,7 +12,19 @@ import { DetailsPage, List, ListHeader, ColHead, ResourceRow } from '../factory/
 
 
 const BaremetalHostDetails = props => {
-  const { metadata, spec, status } = props.bmh;
+  const { metadata } = props.bmh;
+
+  const spec = _.defaults(props.bmh.spec, {
+    online: false,
+  });
+
+  const status = _.defaults(props.bmh.status, {
```

Comment:
cool function, didn't know about it. IMO, I still prefer the selectors approach, i.e. get method with the default parameter, because it doesn't change the underlying object. Changing it, is probably ok, but it might introduce some side-effects in the future